src/rapidhtml/tags.py

```diff
@@ -121,11 +122,53 @@ def __init__(
             )
 
         self.__closing_tag = f"</{self.tag}>" if not self.__self_closing else "/>"
+        self.__uuid: str = (
+            uuid4().hex
+        )  # A unique ID used to detect cyclical references of tags
 
     def __init_subclass__(cls, self_closing: bool = False, **kwargs) -> None:
         super().__init_subclass__(**kwargs)
         cls.__self_closing = self_closing
 
+    @property
+    def _uuid(self) -> str:
+        return self.__uuid
+
+    def validate_cyclical_references(
+        self, *new_tags: "BaseTag", _parents=None
+    ) -> typing.Literal[True]:
+        """
+        Validates if there are any cyclical references within the tags.
+
+        Args:
+            *new_tags (BaseTag): Variable number of new tags to be checked for cyclical references.
+            _parents (List[str], optional): List of parent UUIDs. For internal recursive use only.
+
+        Raises:
+            ValueError: If a cyclical reference is detected.
+
+        Returns:
+            bool: True if no cyclical references are found.
+        """
+
+        parents = _parents or []
+
+        if self.__uuid in parents:
+            raise ValueError("Cyclical reference detected!")
```

Comment:
I'm thinking we should get a `rapidhtml.errors` started and create a `CyclicalReferenceError` to raise here. Admittedly it doesn't serve much of a purpose other than just looking cool and getting specific with thrown errors

Reply:
I've added a custom exception block. It's more stylistic for sure but a nice one to have